contents/testing/integration-testing.md

```diff
@@ -0,0 +1,224 @@
+
+Imagine that you have built an online web ordering system and a sample architecture diagram of your application is shown as follows:
+
+<center>
+
+![Sample architecture diagram](integration-test/sample-architecture.png "Sample architecture diagram")
+</center>
+
+When focusing on the food ordering aspect of this system, a unit test on the `Order Summary` could look like this:
+```
+@Test
+public void isOrderSummaryCorrect(OrderSummary orderSummary) {
+	assertTrue(orderSummary.contains("cheese burger"));
+}
+```
+
+After unit testing of individual modules are completed, an integration test on `Order Food` and `Order Summary` could look like this:
+
+```
+public class FoodOrderingIntegrationTest {
+	public void testFoodOrderRecorded() {
+		FoodOrder foodOrder1 = new FoodOrder("cheese burger");
+		FoodSummary foodSummary = new FoodSummary();
+		foodSummary.add(foodOrder);
+		
+		isOrderSummaryCorrect(foodSummary);
+	}
+}
+```
+
+The system test is used to ensure that your entire application has the necessary working functionalities such as ordering, paying and generate reports. It usually involve different scenarios of using the entire system like this:
+
+- A customer ordered 3 items, made payment using his E-Wallet and generated a report to save the order record.
+
+## Why is integration testing important?
+
+In a software application, each software module is usually designed and unit tested by different programmers.
+Since these programmers may work in isolation and have different understandings of the software requirements, integration testing is necessary to ensure that software modules work in unity and expose any faults in the interaction between different units.
+
+Suppose you are developing the `Order Food` component as mentioned previously and you allow each user to submit up to 10 orders each time, if the `Order Summary` component can only record a maximum of 5 orders each time, 5 orders by different users could be lost, resulting in problems such as **data loss**, **inaccurate results** and even **system errors**.
+Integration tests find these discrepancies before any serious damage is done.
+
+Although system testing also help to uncover faults between modules, integration testing is still highly encouraged as it is **easier to test**, **isolate failures** and **run faster** than a system test.
+Many integration problems can be resolved before they complicate further in your future development, making developers lives more difficult.
+More advantages of integration testing before system testing can be found [here](https://bitwaretechnologies.com/integration-testing-important-software-testing-life-cycle/).
+
+Hence, integration testing is an important process which should be applied in the development of most softwares products such as web, mobile and desktop applications. 
+Concepts shared in this article can be adopted by other different software applications.
```

Comment:
Perhaps you may want to specify what type of application this article targets. However, I feel that it need not be explicitly stated since the article is fairly general.